esp32_s2_kaluga_kit/esp32_s2_kaluga_kit.c

```diff
@@ -1,24 +1,32 @@
 /*
- * SPDX-FileCopyrightText: 2021-2022 Espressif Systems (Shanghai) CO LTD
+ * SPDX-FileCopyrightText: 2021-2023 Espressif Systems (Shanghai) CO LTD
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
 #include "driver/spi_master.h"
+#include "driver/i2c.h"
 #include "esp_err.h"
 #include "esp_log.h"
 #include "esp_check.h"
 #include "esp_lcd_panel_io.h"
 #include "esp_lcd_panel_vendor.h"
 #include "esp_lcd_panel_ops.h"
+#include "esp_spiffs.h"
 
 #include "bsp/esp32_s2_kaluga_kit.h"
 #include "bsp/display.h"
 #include "esp_lvgl_port.h"
+#include "esp_codec_dev_defaults.h"
 #include "bsp_err_check.h"
 
 static const char *TAG = "Kaluga";
 
+static bool i2c_initialized = false;
+
+// Function prototypes
+const audio_codec_data_if_t *bsp_audio_init(void);
```

Comment:
I think, that we should keep it in public API (for make BSP independent on `esp_codec_dev` - something like displays)